boards/BMS/Src/Io/Io_LTC6813/Io_NewLTC6813.c

```diff
@@ -112,61 +123,86 @@ static const uint16_t pec15_lut[PEC15_LUT_SIZE] = {
     0x8BA7, 0x4E3E, 0x450C, 0x8095
 };
 
-// Commands used to write to the configuration registers
-static const uint16_t cfg_reg_cmds[NUM_OF_CFG_REGISTERS] = {
-    [CONFIG_REG_A] = WRCFGA,
-    [CONFIG_REG_B] = WRCFGB,
-};
-
-// Default configuration registers settings
-static const uint8_t default_cfg_reg[NUM_OF_CFG_REGISTERS]
-                                    [NUM_REG_GROUP_BYTES] = {
-                                        [CONFIG_REG_A] = DEFAULT_CFGRA_CONFIG,
-                                        [CONFIG_REG_B] = DEFAULT_CFGRB_CONFIG,
-                                    };
+/**
+ * Configure DCC bits used to control cell discharge
+ * @param tx_cfg Buffer containing data to write to the configuration register
+ * @param curr_cfg_reg The current configuration register to configure
+ * @param curr_segment The current segment to configure
+ * @param min_cell_segment The minimum cell voltage segment location
+ * @param min_cell_index The minimum cell voltage index
+ */
+static void Io_ConfigureDccBits(
+    uint8_t tx_cfg[NUM_OF_ACCUMULATOR_SEGMENTS][NUM_REG_GROUP_PACKET_BYTES],
+    uint8_t curr_cfg_reg,
+    uint8_t curr_segment,
+    uint8_t min_cell_segment,
+    uint8_t min_cell_index);
+
+/**
+ * A function called to prepare data written to configuration registers
+ * @param tx_cfg Buffer containing data to write to the configuration register
+ * @param curr_cfg_reg The current configuration register to configure
+ */
+static void Io_PrepareTxCfg(
+    uint8_t tx_cfg[NUM_OF_ACCUMULATOR_SEGMENTS][NUM_REG_GROUP_PACKET_BYTES],
+    uint8_t curr_cfg_reg);
 
 static void Io_ConfigureDccBits(
-    uint8_t tx_cfg[NUM_OF_ACCUMULATOR_SEGMENTS][NUM_REG_GROUP_BYTES],
+    uint8_t tx_cfg[NUM_OF_ACCUMULATOR_SEGMENTS][NUM_REG_GROUP_PACKET_BYTES],
+    uint8_t curr_cfg_reg,
+    uint8_t curr_segment,
+    uint8_t min_cell_segment,
+    uint8_t min_cell_index)
+{
+    // Get dcc bits to write for the current segment. If this is the lowest
+    // cell, set DCC bits for the given segment
+    const uint16_t dcc_bits =
+        (curr_segment == (NUM_OF_ACCUMULATOR_SEGMENTS - min_cell_segment - 1))
+            ? (uint16_t)(~(SET_MIN_CELL_DCC_BIT(min_cell_index)))
+            : SET_ALL_DCC_BITS;
```

Comment:
If I'm getting this correctly, if the lowest cell voltage is in the current segment, set the DCC bits for all cells except the cell with the lowest voltage. Otherwise, set all DCC bits for that cell. 

Therefore, could we rename `SET_MIN_CELL_DCC_BIT(min_cell_index)` to be something like `SET_DCC_BITS_EXCEPT_MIN_CELL(min_cell_index)`?

The macro definition would be: 
` #define SET_DCC_BITS_EXCEPT_MIN_CELL(_min_cell_index) ~((uint16_t)(1U << (index)))` 

I think this would make more sense from the reader's perspective.

Reply:
yeah good point ill make this change

Reply:
Furthermore, what do you think of adding an inline function for readability? Your call i.e.

static inline IsMinCellWithinCurrentSegment(uint8_t curr_segment, uint8_t min_cell_segment)
{
   return  (curr_segment == (NUM_OF_ACCUMULATOR_SEGMENTS - min_cell_segment - 1); 
}

Reply:
This implementation will likely change, so i wouldn't spend too much time on this. The idea was that to show that we can configure DCC bits and run a simple algo discharging to the min cell